client/src/controllers/MessagesController.ts

```diff
@@ -0,0 +1,77 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Adds the ability for a controlled element to pick messages from template,
+ * add/stack (depending upon clear:true/false) them within the provided unordered list container
+ * when called.
+ *
+ * @example
+ * <div data-controller="w-messages"
+ *      data-action="w-messages:add@document->w-messages#add"
+        data-w-messages-added-class="new"
+        data-w-messages-show-class="appear"
+ * >
+      <ul data-w-messages-target="messages"></ul>
+      <template data-w-messages-target="templates">
+        <li data-message-status="error/success"><span></span></li>
+      </template>
+ * </div>
+ */
+export class MessagesController extends Controller<HTMLFormElement> {
+  static classes = ['added', 'show'];
+  static targets = ['messages', 'templates'];
+  static values = {
+    showDelay: { default: 100, type: Number },
+  };
+
+  declare readonly hasMessagesTarget: boolean;
+  declare readonly hasAddedClass: boolean;
+  declare readonly hasShowClass: boolean;
+  declare readonly hasTemplatesTarget: boolean;
+  declare messagesTarget: HTMLElement;
+  declare readonly addedClass: string;
+  declare readonly showClass: string;
+  declare readonly showDelayValue: number;
+  declare templatesTarget: HTMLTemplateElement;
+
+  add(
+    event?: CustomEvent<{
+      /** Message status level, based on Django's message types. */
+      status?: 'success' | 'error' | 'warning' | string;
+      /** Content for the message, HTML not supported. */
+      text?: string;
+      /** flag for clearing or stacking messages */
+      clear?: boolean;
+    }>,
+  ) {
+    const {
+      status = 'other',
+      text: statusText,
+      clear = false,
+    } = event?.detail || {};
+    if (!status && !statusText) return;
```

Comment:
```suggestion
  add(
    event?: CustomEvent<{
      /** Flag for clearing or stacking messages */
      clear?: boolean;
      /** Message status level, based on Django's message types. */
      status?: 'success' | 'error' | 'warning' | string;
      /** Content for the message, HTML not supported. */
      text?: string;
    }>,
  ) {
    const {
      clear = false,
      status = 'success',
      text = '',
    } = event?.detail || {};

    if (!status && !text) return;

```

Bit of clean up, sort params but also let's not rename to `statusText`, there is no real need for this.